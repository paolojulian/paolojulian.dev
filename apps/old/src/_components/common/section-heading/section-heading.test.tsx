import { render } from '@testing-library/react';
import SectionHeading from './section-heading';
import { ComponentProps } from 'react';

describe('TESTING SectionHeading component', () => {
  describe('GIVEN valid props', () => {
    const mockData: ComponentProps<typeof SectionHeading> = {
      title: 'Sample Heading',
      RightContent: <div>Right Content</div>,
    };

    describe('WHEN the SectionHeading component is rendered in snapshot testing', () => {
      it('THEN it should match the snapshot', () => {
        const { asFragment } = renderSectionHeading(mockData);
        expect(asFragment()).toMatchSnapshot();
      });
    });

    describe('WHEN the SectionHeading component is rendered', () => {
      it('THEN it should render the title correctly', () => {
        const { getByText } = renderSectionHeading(mockData);
        const title = getByText(mockData.title);
        expect(title).toBeInTheDocument();
      });

      it('THEN it should render the RightContent correctly', () => {
        const { getByText } = renderSectionHeading(mockData);
        const rightContent = getByText('Right Content');
        expect(rightContent).toBeInTheDocument();
      });
    });
  });
  describe('GIVEN an empty RightContent', () => {
    const mockData: ComponentProps<typeof SectionHeading> = {
      title: 'Sample Heading',
    };

    describe('WHEN the SectionHeading component is rendered in snapshot testing', () => {
      it('THEN it should match the snapshot', () => {
        const { asFragment } = renderSectionHeading(mockData);
        expect(asFragment()).toMatchSnapshot();
      });
    });
  });
});

function renderSectionHeading(props: ComponentProps<typeof SectionHeading>) {
  return render(<SectionHeading {...props} />);
}
