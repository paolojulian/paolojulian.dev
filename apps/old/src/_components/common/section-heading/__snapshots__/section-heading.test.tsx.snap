// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TESTING SectionHeading component GIVEN an empty RightContent WHEN the SectionHeading component is rendered in snapshot testing THEN it should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="border-b-2 border-new-white text-new-white flex justify-between items-end md:p-2"
  >
    <h2
      class="uppercase font-medium font-comfortaa tracking-wide tracking-widest md:tracking-[0.375rem] text-base"
    >
      Sample Heading
    </h2>
    <span />
  </div>
</DocumentFragment>
`;

exports[`TESTING SectionHeading component GIVEN valid props WHEN the SectionHeading component is rendered in snapshot testing THEN it should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="border-b-2 border-new-white text-new-white flex justify-between items-end md:p-2"
  >
    <h2
      class="uppercase font-medium font-comfortaa tracking-wide tracking-widest md:tracking-[0.375rem] text-base"
    >
      Sample Heading
    </h2>
    <span>
      <div>
        Right Content
      </div>
    </span>
  </div>
</DocumentFragment>
`;
